docs/build/guides/conventions/workspace.mdx

```diff
@@ -0,0 +1,383 @@
+
+```bash
+cd $PROJECT_ROOT
+stellar contract init . --name adder_interface
+stellar contract init . --name add_extra_contract
+```
+
+First, let's create an interface and change our existing implementation to use this interface:
+
+```rust
+// contracts/adder_interface/src/lib.rs
+#![no_std]
+
+use soroban_sdk::contractclient;
+
+#[contractclient(name = "ContractAClient")]
+pub trait ContractAInterface {
+    fn add(x: u32, y: u32) -> u32;
+}
+```
+
+To use the interface definition our workspace members will now have an `adder_interface` as a dependency:
+
+```toml
+# $PROJECT_PATH/Cargo.toml
+# <...>
+[workspace.dependencies]
+soroban-sdk = "21.0.0"
+adder-interface = { path = "contracts/adder_interface" }
+# <...>
+```
+
+```toml
+# $PROJECT_PATH/contracts/add_contract/Cargo.toml
+# <...>
+[dependencies]
+soroban-sdk = { workspace = true }
+adder-interface = {workspace = true}
+# <...>
+```
+
+```toml
+# $PROJECT_PATH/contracts/main_contract/Cargo.toml
+# <...>
+[dependencies]
+soroban-sdk = { workspace = true }
+adder-interface = {workspace = true}
+# <...>
+```
+
+```toml
+# $PROJECT_PATH/contracts/add_extra_contract/Cargo.toml
+# <...>
+[dependencies]
+soroban-sdk = { workspace = true }
+adder-interface = {workspace = true}
+# <...>
+```
+
+And change lib type of `adder_interface` crate:
+
+```toml
+# $PROJECT_PATH/contracts/adder_interface/Cargo.toml
+# <...>
+[lib]
+crate-type = ["rlib"]
+# <...>
+```
+
+```rust
+# $PROJECT_PATH/contracts/adder_interface/src/lib.rs
+#![no_std]
+
+use soroban_sdk::contractclient;
+
+#[contractclient(name = "AdderClient")]
+pub trait Adder {
+    fn add(x: u32, y: u32) -> u32;
+}
+
+```
+
+```rust
+// contracts/add_contract/src/lib.rs
+#![no_std]
+use soroban_sdk::{contract, contractimpl};
+use adder_interface::Adder;
+
+#[contract]
+pub struct ContractAdd;
+
+#[contractimpl]
+impl Adder for ContractAdd {
+    fn add(x: u32, y: u32) -> u32 {
+        x.checked_add(y).expect("no overflow")
+    }
+}
+
+```
+
+```rust
+// contracts/main_contract/src/lib.rs
+#![no_std]
+
+use soroban_sdk::{contract, contractimpl, Address, Env};
+use adder_interface::AdderClient;
+
+#[contract]
+pub struct ContractMain;
+
+#[contractimpl]
+impl ContractMain {
+    pub fn add_with(env: Env, contract: Address, x: u32, y: u32) -> u32 {
+        let client = AdderClient::new(&env, &contract);
+        client.add(&x, &y)
+    }
+}
+
+mod test;
+```
+
+And update test imports:
+
+```rust
+// contracts/main_contract/src/test.rs
+#![cfg(test)]
+
+use crate::{ContractMain, ContractMainClient};
+use soroban_sdk::Env;
+
+mod contract_add {
+    soroban_sdk::contractimport!(
+        file = "../../target/wasm32-unknown-unknown/release/my_contract.wasm"
+    );
+}
+// <...>
+```
+
+As the final step we can create an alternative `Adder` implementation that adds an extra 1:
+
+```rust
+// contracts/add_extra_contract/src/lib.rs
```

Comment:
You can set the title on the code block with:
````suggestion
```rust title="contracts/add_extra_contract/src/lib.rs"
````